### PR TITLE
Make `PresentMode` configurable.

### PR DESCRIPTION
**Pull request by Yatekii**

### CODE REVIEW

File: wgpu-native/src/device.rs [hunk 1]
```diff
@@ -1731,7 +1731,7 @@ pub fn device_create_swap_chain(
         desc.height,
         conv::map_texture_format(desc.format),
         num_frames, //TODO: configure?
-    );
+    ).with_mode(desc.present_mode);
```
Comment (kvark) [inline wgpu-native/src/device.rs]: let's just do it like `composite_alpha` - with a simple assignment

File: wgpu-native/src/swap_chain.rs [hunk 2]
```diff
@@ -77,13 +77,30 @@ pub struct SwapChain<B: hal::Backend> {
     pub(crate) command_pool: hal::CommandPool<B, hal::General>,
 }
 
+#[repr(C)]
+#[derive(Clone, Debug)]
```
Comment (kvark) [inline wgpu-native/src/swap_chain.rs]: let's also derive `Copy`

File: wgpu-native/src/device.rs [hunk 3]
```diff
@@ -1734,6 +1734,7 @@ pub fn device_create_swap_chain(
     );
     //TODO: check for supported
     config.composite_alpha = hal::window::CompositeAlpha::OPAQUE;
+    config.present_mode = desc.present_mode.clone().into();
```
Comment (kvark) [inline wgpu-native/src/device.rs]: let's just do a `match` here without implementing `into()`

**Comment by Yatekii** [comment]
The 

> Thank you for addressing the concerns!
> Where did wgpu.h go?

Well I can't build the C examples as of now still ... I just made an intermediate commit as it builds for Rust only.
Dunno how to fix it.
But ofc that gets pushed to the PR.

File: wgpu-native/src/device.rs [hunk 4]
```diff
@@ -1734,6 +1734,10 @@ pub fn device_create_swap_chain(
     );
     //TODO: check for supported
     config.composite_alpha = hal::window::CompositeAlpha::OPAQUE;
+    config.present_mode = match &desc.present_mode {
```
Comment (kvark) [inline wgpu-native/src/device.rs]: shouldn't need the "&"

Comment (kvark) [reply wgpu-native/src/device.rs]: this is still not fixed?

**Comment by kvark** [comment]
bors try

**Comment by kvark** [comment]
Bors r+

**Pull request merged**